README.md

```diff
@@ -162,6 +162,84 @@ $sumThree(7, 42, 666);
 
 All invocations will yield the same result.
 
+#### Avoid billion dollar mistake by using `Option`
+
+`Option` type encapsulates value, which may or may not exist. If you are not familiar with concept of `Option` (also called `Maybe` in some languages), think of `ArrayList` which is either empty or has single item inside.
+
+Value which exists is represented by instance of `Some` class, whereas missing one is `None`.
+
+```php
+use Bonami\Collection\Option;
+
+$somethingToEat = Option::some("ice cream");
+$nothingToSeeHere = Option::none();
+```
+
+The good thing is that we can operate on `Some` & `None` the same way: 
+
+```php 
+use Bonami\Collection\Option;
+
+$somethingToEat = Option::some("ice cream");
+$nothingToSeeHere = Option::none();
+
+$iLikeToEat = fn (string $food): string => "I like to eat tasty {$food}!"; 
+
+$somethingToEat->map($iLikeToEat); // Will map to string "I like to eat tasty ice cream!" wrapped in `Some` instance
+$nothingToSeeHere->map($iLikeToEat); // `None`, wont be mapped and will stay the same
+
+``` 
+ 
+We can use `Option` as better and more safe alternative to nullable values since handling of `null` may easily become cumbersome. Consider following code example:
+
+```php
+use Bonami\Collection\Option;
+
+function getUserEmailById(int $id): ?string {
+    $usersDb = [
+        1 => "john@foobar.baz",
+        2 => "paul@foobar.baz",
+    ];
+    return $usersDb[$id] ?? null;
+} 
+function getAgeByUserEmail(string $email): ?int {
+    $ageDb = [
+        "john@foobar.baz" => 66,
+        "diego@hola.esp" => 42,
+    ];
+    return $ageDb[$email] ?? null;
+}
+
+function printUserAgeByIdTheNullWay(int $id): void {
+    $email = getUserEmailById($id);
+    $age = null;
+    if ($email !== null) {
+        $age = getAgeByUserEmail($email);
+   
+    }
+    if ($age === null) {
+        print "Dont know age of user with id {$id}";
+    } else {
+        print "Age of user with id {$id} is {$age}";
+    }   
+     
+}
+
```

Comment:
I would split this code snippet into two and first one I would label with "classic approach with null" and second one "compare with functional approach with Option". Or something like that

Reply:
I used to have it that way :-D 

Will revert it...